core/src/main/java/com/linecorp/armeria/client/retry/HttpStatusBasedRetryStrategy.java

```diff
@@ -68,11 +100,13 @@
             if (headers != null) {
                 final HttpStatus resStatus = headers.status();
                 if (resStatus != null) {
-                    return retryStatuses.stream().anyMatch(
-                            retryStatus -> resStatus.code() == retryStatus.code());
+                    final Backoff backoff = retryStatusesWithBackoff.get(resStatus);
+                    if (backoff != null) {
+                        return Optional.of(backoff);
```

Comment:
Could be: `return Optional.ofNullable(retryStatusesWithBackoff.get(resStatus));`